pkg/auth/providers/common/usermanager.go

```diff
@@ -334,20 +341,14 @@ func (m *userManager) newTokenForKubeconfig(clusterName, tokenName, description,
 	return key, nil
 }
 
-// creates kubeconfig tokens with KubeconfigTokenTTL and regenerates if existing token expired
+// GetKubeconfigToken creates a new token for use in a kubeconfig generated through the CLI.
 func (m *userManager) GetKubeconfigToken(clusterName, tokenName, description, kind, userName string, userPrincipal v3.Principal) (*v3.Token, string, error) {
-
-	tokenTTL, err := tokens.ParseTokenTTL(settings.KubeconfigTokenTTLMinutes.Get())
-	if err != nil {
-		return nil, "", fmt.Errorf("failed to parse setting [%s]: %v", settings.KubeconfigTokenTTLMinutes.Name, err)
-	}
-
-	fullCreatedToken, err := m.newTokenForKubeconfig(clusterName, tokenName, description, kind, userName, tokenTTL, userPrincipal)
+	fullCreatedToken, err := m.newTokenForKubeconfig(clusterName, tokenName, description, kind, userName, userPrincipal)
 	if err != nil {
 		return nil, "", err
 	}
 
-	_, createdTokenValue := tokens.SplitTokenParts(fullCreatedToken)
+	tokenName, createdTokenValue := tokens.SplitTokenParts(fullCreatedToken)
```

Comment:
I don't understand this change, won't this not use `tokenName` that was passed in ? Or is it actually the same ?

Reply:
We are reassigning `tokenName` because the above call`newTokenForKubeconfig` now suffixes the passed in `tokenName` value with a random string. For clarity, I could have created a new variable called `randomizedTokenName`.
That being said I think it would be better to change the `newTokenForKubeconfig` function to return a token object since we are doing `tokenLister.Get()` for the token directly afterward.

Reply:
Ahh ok, so randomizedTokenName is now different from the arg tokenName.